torch/utils/data/datapipes/datapipe.py

```diff
@@ -232,6 +237,10 @@ class MapDataPipe(Dataset[T_co], metaclass=_DataPipeMeta):
 
     def __getattr__(self, attribute_name):
         if attribute_name in MapDataPipe.functions:
+            if attribute_name in _deprecated_functional_names:
+                type, kwargs = _deprecated_functional_names[attribute_name]
```

Comment:
Micro performance improvement. Why not split this `_deprecated_functional_names` to `_iter_deprecated_functional_names` and `_map_deprecated_functional_names` to reduce time to search the `attribute_name` in a single dictionary?